spec.bs

```diff
@@ -2305,24 +2302,20 @@ Each {{InterestGroupBiddingScriptRunnerGlobalScope}} has a
 
 </dl>
 
-<div algorithm>
-  The <dfn method for="InterestGroupBiddingScriptRunnerGlobalScope">setBid()</dfn> method steps are:
-
-  1. Set [=this=]'s [=relevant global object=]'s [=InterestGroupBiddingScriptRunnerGlobalScope/bid=]
-    to null.
-  1. Return true.
-</div>
 
 <div algorithm>
 
 To <dfn>convert GenerateBidOutput to generated bid</dfn> given a {{GenerateBidOutput}}
 |generateBidOutput|, an [=interest group=] |ig|, a [=currency tag=] |expectedCurrency|, a [=boolean=]
 |isComponentAuction| and a [=boolean=] |groupHasAdComponents|:
-  1. If |generateBidOutput|["{{GenerateBidOutput/bid}}"] is less than or equal to 0, return failure.

```

Comment:
then I guess this [mojom commen](https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:content/services/auction_worklet/public/mojom/bidder_worklet.mojom;drc=fdfd85f836e0e59c79ed9bf6d527a2b8f7fdeb6e;l=102)t is wrong, since at this level (moment), we're not enforcing it to be positive, right?

Reply:
No, it's correct; since the negative bid values mean "no bid" we just return a null BidderWorkletBidPtr in the impl.
In here, all this dance is because we have to distinguish between:
   setBid({bid: -1}) 
which sets pending bid to "no bid" and something like:
   setBid{{bid: 10})
which throws an exception because a valid bid must have render: set.

... But by the time we're done with this process we no longer care; you can see it in the spec, too, in line 2146 in the revised version --- bids <= get folded right back into failure.